langchain/src/vectorstores/tests/lancedb.test.ts

```diff
@@ -0,0 +1,69 @@
+import { test, expect } from "@jest/globals";
+import { track } from "temp";
+import { connect, Table } from "vectordb";
+import { LanceDB } from "../lancedb.js";
+import { FakeEmbeddings } from "../../embeddings/fake.js";
+import { Document } from "../../document.js";
+
+describe("LanceDB", () => {
+  test("constructor works", async () => {
+    const lancedb = new LanceDB(new FakeEmbeddings(), {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      table: new Table(null as any, "vectors"),
+    });
+    expect(lancedb).toBeDefined();
+  });
+
+  test("should add vectors to the table", async () => {
+    const embeddings = new FakeEmbeddings();
+    const dir = await track().mkdir("lancedb");
```

Comment:
Do we need to import `track` as an another dev-dependency? Could we just remove the temp directory after the test completes?

Reply:
It did not pass our CI, build seemed to fail on old Macs and Windows. It's all good to have it as an integration test IMO.